dot/sync/disjoint_block_set_test.go

```diff
@@ -0,0 +1,331 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package sync
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	testBlock1 = &pendingBlock{
+		hash:   common.Hash{1},
+		number: 1,
+	}
+	testBlock10 = &pendingBlock{
+		hash:   common.Hash{10},
+		number: 10,
+	}
+)
+
+func Test_disjointBlockSet_addBlock(t *testing.T) {
+	t.Parallel()
+
+	type fields struct {
+		limit            int
+		blocks           map[common.Hash]*pendingBlock
+		parentToChildren map[common.Hash]map[common.Hash]struct{}
+		timeNow          func() time.Time
+	}
+	type args struct {
+		block *types.Block
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		err    error
+	}{
+		{
+			name:   "add block beyond capacity",
+			fields: fields{},
+			args: args{block: &types.Block{
+				Header: types.Header{
+					Number: 1,
+				},
+			}},
+			err: errors.New("cannot add block; set is at capacity"),
+		},
+		{
+			name: "add block",
+			fields: fields{
+				limit:            1,
+				blocks:           make(map[common.Hash]*pendingBlock),
+				timeNow:          time.Now,
+				parentToChildren: make(map[common.Hash]map[common.Hash]struct{}),
+			},
+			args: args{block: &types.Block{
+				Header: types.Header{
+					Number: 1,
+				},
+			}},
+			err: nil,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := &disjointBlockSet{
+				limit:            tt.fields.limit,
+				blocks:           tt.fields.blocks,
+				parentToChildren: tt.fields.parentToChildren,
+				timeNow:          tt.fields.timeNow,
+			}
+			err := s.addBlock(tt.args.block)
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func Test_disjointBlockSet_addHeader(t *testing.T) {
+	t.Parallel()
+
+	type fields struct {
+		limit            int
+		blocks           map[common.Hash]*pendingBlock
+		parentToChildren map[common.Hash]map[common.Hash]struct{}
+		timeNow          func() time.Time
+	}
+	type args struct {
+		header *types.Header
+	}
```

Comment:
nit Use `header` directly instead of `args` struct?

Reply:
Updated.